lib/gen_stage.ex

```diff
@@ -1815,22 +1867,31 @@ defmodule GenStage do
   def init({mod, args}) do
     case mod.init(args) do
       {:producer, state} ->
-        init_producer(mod, [], state)
+        init_producer(mod, [], state, nil)
```

Comment:
Instead of passing `nil` around to have a single `init_producer/4` helper function, what if we call `init_producer(mod, [], state)` most of the time and define a different `init_producer/4`, something like:

```elixir
defp init_producer(mod, opts, state, additional_info) do
  with {:ok, stage} <- init_producer(mod, opts, state) do
    {:ok, stage, additional_info}
  end 
end
```

Thoughts?

Reply:
makes sense!